portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -196,4 +203,22 @@ private String getUploadedFileUrl(HttpServletRequest request, String formInputEl
       return imagesService.getServingUrl(options);
     }
   }
+
+  /** Returns the nickname of the user with id, or null if the user has not 
+   * set a nickname. 
+   * @param id user id of the user.
+   */
+  private String getUserNickname(String id) {
```

Comment:
The same function is used by both the DataServlet and NicknameServlet, so should I move it into an external class so that I don't have to repeat the implementation in both servlets? 
That seems to be the better practice to avoid repeated code, but not sure what I would name the external class (since it would really just have one method). Or should I make it a public method in the NicknameServlet?

Reply:
I think either creating a class called like `Nickname` that has public static functions to deal with getting/setting nicknames from the datastore, or changing `getUserNickname()` to be public static in `NicknameServlet` are fine. Maybe try both and see what you like better? Note you will want these functions to be `static` so you don't have to create an instance of the class to call them. In general helper functions are `static` because you don't need to access any member variables.

Reply:
I've made a new Nickname class as you suggested, and put the `getUserNickname` function as a static method in there. I think the function is not really related to the the servlet's purpose, so I think a new class is the better way to go. Thankyou!